src/CurrentPlugin.php

```diff
@@ -42,20 +42,20 @@ function read_block(int $offset) : string
 
     /**
      * Allocates a block of memory in the plugin's memory and returns the offset.
-     * 
+     *
      * @param int $size Size of the block to allocate in bytes.
      */
-    function allocate_block(int $size) : int
+    public function allocate_block(int $size): int
```

Comment:
do these need to be public? These are fairly low-level methods that we don't want people calling in normal operation.

Reply:
Actually I am not completely sure what does these function behind the scene. As far as I checked it is used in HostFunctions.php. If you call these methods in the HostFunctions they should be public, if you don't you can set them as private.

Reply:
I've pushed some code and I changed the visibility of unused functions in CurrentPlugin.php. You can guide me about updating visibility of specific methods.